k9mail/src/test/java/org/openintents/openpgp/util/ShadowOpenPgpAsyncTask.java

```diff
@@ -0,0 +1,25 @@
+package org.openintents.openpgp.util;
+
+
+import java.util.concurrent.Executor;
+
+import android.content.Intent;
+import android.os.AsyncTask;
+
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+import org.robolectric.annotation.RealObject;
+import org.robolectric.shadows.ShadowAsyncTask;
+
+
+@Implements(OpenPgpApi.OpenPgpAsyncTask.class)
+public class ShadowOpenPgpAsyncTask extends ShadowAsyncTask<Void, Integer, Intent> {
```

Comment:
Shadows are basically like mocks that stub out bits of Android and allow you to test without a device. But they don't always do what you want. In this case, in order to simulate tasks being run on the executor you have to replace the shadow implementation (this specific case is covered here: http://stackoverflow.com/questions/26290867/asynctask-test-doesnt-execute ).